easybuild/easyblocks/i/ipp.py

```diff
@@ -1,77 +1,82 @@
-# the Hercules foundation (http://www.herculesstichting.be/in_English)
-# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#  Copyright (C) 2015 IT4Innovations
+#  Lumir Jasiok
+#  lumir.jasiok@vsb.cz
+#  http://www.it4i.cz
 #
-# http://github.com/hpcugent/easybuild
 #
-# EasyBuild is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation v2.
 #
-# EasyBuild is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
-##
-"""
-EasyBuild support for installing the Intel Performance Primitives (IPP) library, implemented as an easyblock
 
-@author: Stijn De Weirdt (Ghent University)
-@author: Dries Verdegem (Ghent University)
-@author: Kenneth Hoste (Ghent University)
-@author: Pieter De Baets (Ghent University)
-@author: Jens Timmerman (Ghent University)
 """
-from distutils.version import LooseVersion
+EasyBuild support for building and installing Intel Performance Primitives (IPP) library, implemented as an easyblock. 
+@author: Lumir Jasiok (IT4Innovations)
+"""
 
-from easybuild.easyblocks.generic.intelbase import IntelBase, ACTIVATION_NAME_2012, LICENSE_FILE_NAME_2012
+import os
+import platform
 
+from easybuild.easyblocks.generic.intelbase import IntelBase
+from easybuild.tools.run import run_cmd
 
 class EB_ipp(IntelBase):
     """
-    Support for installing Intel Integrated Performance Primitives library
+    Support for installing Intel Performance Primitives
     """
+
     def install_step(self):
+        """Actual installation
+        
+        - create silent.cfg
+        - run install.sh
         """
-        Actual installation
-        - create silent cfg file
-        - execute command
-        """
-        silent_cfg_names_map = None
-
-        if LooseVersion(self.version) < LooseVersion('8.0'):
-            silent_cfg_names_map = {
-                'activation_name': ACTIVATION_NAME_2012,
-                'license_file_name': LICENSE_FILE_NAME_2012,
-            }
+        
+        machine = platform.machine()
+        if machine == "i386":
+            self.arch = "ia32"
+        else:
+            self.arch = "intel64"
 
-        super(EB_ipp, self).install_step(silent_cfg_names_map=silent_cfg_names_map)
+        silent_cfg = """
+ACCEPT_EULA=accept
+CONTINUE_WITH_OPTIONAL_ERROR=yes
+PSET_INSTALL_DIR=%s
+CONTINUE_WITH_INSTALLDIR_OVERWRITE=yes
+COMPONENTS=ALL
+PSET_MODE=install
+ACTIVATION_LICENSE_FILE=%s
+ACTIVATION_TYPE=license_file
+PHONEHOME_SEND_USAGE_DATA=no
+SIGNING_ENABLED=yes
+ARCH_SELECTED=%s
```

Comment:
I don't see why this cannot be handled in the `intelbase` block? It's all generic.